c/parallel/src/jit_templates/CMakeLists.txt

```diff
@@ -5,28 +5,62 @@ set(cpp_entry "${CMAKE_CURRENT_LIST_DIR}/jit_entry.h")
 
 file(MAKE_DIRECTORY "${CMAKE_CURRENT_BINARY_DIR}")
 
+# The pre-processor (PP) command differs between MSVC and others.
+set(PP_CMD ${CMAKE_CXX_COMPILER})
+set(PP_DEPFILE_KW)
+set(PP_ARGS)
+
+if(MSVC)
+  list(APPEND PP_ARGS /D_CCCL_C_PARALLEL_JIT_TEMPLATES_PREPROCESS)
+else()
+  list(APPEND PP_ARGS -D_CCCL_C_PARALLEL_JIT_TEMPLATES_PREPROCESS)
+endif()
+
+if(MSVC)
+  file(TO_NATIVE_PATH "${jit_template}" JIT_TEMPLATE_NATIVE)
+
+  list(APPEND PP_ARGS
+    /nologo
+    /TP
+    /P
+    /Fi${JIT_TEMPLATE_NATIVE}
+    "${cpp_entry}"
+  )
+else()
+  list(APPEND PP_ARGS
+    -E "${cpp_entry}"
+    -o "${jit_template}"
+    -MD -MT "${jit_template}"
+    -MF "${jit_template_depfile}"
+  )
+  set(PP_DEPFILE_KW DEPFILE "${jit_template_depfile}")
+endif()
+
 add_custom_command(
-    OUTPUT "${jit_template}"
-    DEPENDS "${cpp_entry}"
-    DEPFILE "${jit_template_depfile}"
-    COMMAND "${CMAKE_CXX_COMPILER}" "${cpp_entry}" -E -o "${jit_template}"
-        -D_CCCL_C_PARALLEL_JIT_TEMPLATES_PREPROCESS
-        -MD -MT "${jit_template}" -MF "${jit_template_depfile}"
+  OUTPUT "${jit_template}"
+  DEPENDS "${cpp_entry}"
+  COMMAND "${PP_CMD}" ${PP_ARGS}
+  ${PP_DEPFILE_KW}
```

Comment:
MSVC supports `/sourceDependencies` to generate dependency files: https://learn.microsoft.com/en-us/cpp/build/reference/sourcedependencies-directives?view=msvc-170

Perhaps we would need to give up dependency tracking on Windows.